centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -259,10 +259,54 @@ public function existsByAccessGroups(int $hostCategoryId, array $accessGroups):
         foreach ($concat->retrieveBindValues() as $param => [$value, $type]) {
             $statement->bindValue($param, $value, $type);
         }
-        $statement->bindValue(':hostCategoryId', $hostCategoryId, \PDO::PARAM_INT);
+        $statement->bindValue(':hostCategoryId', $hostCategoryId, \PDO::PARAM_STR);
 
         $statement->execute();
 
         return (bool) $statement->fetchColumn();
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function existsByName(string $hostCategoryName): bool
+    {
+        $this->info('Check existance of host category with name ' . $hostCategoryName);
+
+        $request = $this->translateDbName(
+            'SELECT hc.hc_id FROM `:db`.hostcategories hc WHERE hc.hc_name = :hostCategoryName'
```

Comment:
Use `SELECT 1` to avoid the theorically `0` to `false` cast.
```suggestion
            'SELECT 1 FROM `:db`.hostcategories hc WHERE hc.hc_name = :hostCategoryName'
```